packages/lesswrong/server/resolvers/postResolvers.ts

```diff
@@ -362,18 +375,46 @@ export type PostIsCriticismRequest = {
 
 addGraphQLResolvers({
   Query: {
-    async UserReadHistory(root: void, args: {limit: number|undefined}, context: ResolverContext) {
-      const { currentUser, repos } = context
+    async UserReadHistory(
+      root: void,
+      args: {
+        limit: number,
+        filter?: FilterPostsForReview,
+        sort?: {
+          karma?: boolean,
+        },
+      }, context: ResolverContext) {
+      const {currentUser, repos} = context
       if (!currentUser) {
         throw new Error('Must be logged in to view read history')
       }
 
-      const posts = await repos.posts.getReadHistoryForUser(currentUser._id, args.limit ?? 10)
-      const filteredPosts = accessFilterMultiple(currentUser, Posts, posts, context);
+      const posts = await repos.posts.getReadHistoryForUser(currentUser._id, args.limit ?? 10, args.filter, args.sort)
+      const filteredPosts = accessFilterMultiple(currentUser, Posts, posts, context)
       return {
         posts: filteredPosts,
       }
     },
+    async PostsUserCommentedOn(
+      root: void,
+      {limit = 20, filter, sort}: { 
```

Comment:
I am not 100% sure providing a default limit via destructing-default-assignment works if missing args end up being `null` rather than `undefined`

Reply:
it doesn't =\